### PR TITLE
New templates for endpoint, handler, and saga

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

File: src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+          "choice": "SQS",
+          "description": "Amazon SQS"
+        },
+        {
+          "choice": "RabbitMQ",
+          "description": "RabbitMQ"
+        },
+        {
+          "choice": "SQL",
+          "description": "Sql Transport"
```
Comment (ramonsmits) [inline src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json:68]:
```suggestion
          "description": "Microsoft SQL Server Transport"
```

File: src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+          "choice": "RavenDB",
+          "description": "RavenDB"
+        },
+        {
+          "choice": "MongoDB",
+          "description": "MongoDB"
+        },
+        {
+          "choice": "DynamoDB",
+          "description": "DynamoDB"
```
Comment (ramonsmits) [inline src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json:104]:
```suggestion
          "description": "Amazon DynamoDB"
```

File: src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+          "choice": "LearningPersistence",
+          "description": "Learning Persistence"
+        },
+        {
+          "choice": "SQL",
+          "description": "SQL (MSSQL, PostgreSQL, MySQL, or Oracle)"
+        },
+        {
+          "choice": "CosmosDB",
+          "description": "Cosmos DB"
```
Comment (ramonsmits) [inline src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json:88]:
```suggestion
          "description": "Azure Cosmos DB"
```

File: src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      "defaultValue": "net7.0"
+    },
+    "transport": {
+      "type": "parameter",
+      "description": "The message queue (transport) to use.",
+      "datatype": "choice",
+      "choices": [
+        {
+          "choice": "LearningTransport",
+          "description": "Learning Transport"
```
Comment (ramonsmits) [inline src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json:52]:
```suggestion
          "description": "Learning Transport (non-production)"
```

File: src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      "defaultValue": "LearningTransport"
+    },
+    "persistence": {
+      "type": "parameter",
+      "description": "Where to store data. This should be the same as your business data.",
+      "datatype": "choice",
+      "choices": [
+        {
+          "choice": "LearningPersistence",
+          "description": "Learning Persistence"
```
Comment (ramonsmits) [inline src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json:80]:
```suggestion
          "description": "Learning Persistence (non production)"
```

**Review by ramonsmits — COMMENTED** [review]
Suggestions for the drop down list values.

File: src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+      "type": "parameter",
+      "description": "The message queue (transport) to use.",
+      "datatype": "choice",
+      "choices": [
+        {
+          "choice": "LearningTransport",
+          "description": "Learning Transport"
+        },
+        {
+          "choice": "AzureServiceBus",
```
Comment (andreasohlund) [inline src/ParticularTemplates/NServiceBusEndpoint/.template.config/template.json:55]: Is ASQ missing from the list?

File: src/ParticularTemplates/NServiceBusEndpoint/Program.cs [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+#endif
+#if (persistence == "RavenDB")
+using Raven.Client.Documents;
+#endif
+
+namespace ProjectName
+{
+    static class Program
+    {
+        public static void Main(string[] args)
```
Comment (andreasohlund) [inline src/ParticularTemplates/NServiceBusEndpoint/Program.cs:26]: Should we use the `Task Main(..)` version just to me more modern?

**Review by andreasohlund — APPROVED** [review]
Side note: All those approval file really make it cumbersome to make and review changes :(

File: src/ParticularTemplates/NServiceBusEndpoint/Program.cs (hunk 7 above)
Comment (andreasohlund) [reply src/ParticularTemplates/NServiceBusEndpoint/Program.cs:26]: There is a `RunAsync` but if MS does this I guess we should follow suit

**Pull request merged**